### PR TITLE
Add bearer support for authorization headers

### LINKED ISSUE #19752: websocket token report etcdserver: invalid auth token
### Bug report criteria

- [x] This bug report is not security related, security issues should be disclosed privately via [etcd maintainers](mailto:etcd-maintainers@googlegroups.com).
- [x] This is not a support request or question, support requests or questions should be raised in the etcd [discussion forums](https://github.com/etcd-io/etcd/discussions).
- [x] You have read the etcd [bug reporting guidelines](https://github.com/etcd-io/etcd/blob/main/Documentation/contributor-guide/reporting_bugs.md).
- [x] Existing open issues along with etcd [frequently asked questions](https://etcd.io/docs/latest/faq) have been checked and this is not a duplicate.

### What happened?

when i use etcd api for kotlin ktot non grpc. I find problem,I use websocket for watch key update.anything is hard.I was commit a discussion for help.but,i do not have time.i download etcd source from github,and use github Copilot help me read code source.wow, it help me for how use token by websocket in etcd api.but , it use one problem,the websocket proxy add 'bearer' scheme for token in Authorization header,but grpc not have,so it report rpc error:
`"cancel_reason":"rpc error: code = Unauthenticated desc = etcdserver: invalid auth token"` 
and token take code is:
```
httpClient.ws(method = HttpMethod.Get, host = "localhost", port = 2379, path = "/v3/watch",request = {
        // 添加认证 header
        header(HttpHeaders.SecWebSocketProtocol,"Bearer, $token")
    })
```

### What did you expect to happen?

I want the etcd authentication to work properly regardless of what HTTP Authorization header scheme I use.

### How can we reproduce it (as minimally and precisely as possible)?

etcd enable auth, use websocket client call etcd watch endpoint for ws, first auth for get token , and take token for ws by etcd code source example
` Sec-Websocket-Protocol: Bearer, ${token}`

### Anything else we need to know?

_No response_

### Etcd version (please run commands below)

<details>

```console
$ etcd --version
# 3.5.17

$ etcdctl version
# paste output here
```

</details>

### Etcd configuration (command line flags or environment variables)

<details>

# paste your configuration here

</details>

### Etcd debug information (please run commands below, feel free to obfuscate the IP address or FQDN in the output)

<details>

```console
$ etcdctl member list -w table
# paste output here

$ etcdctl --endpoints=<member list> endpoint status -w table
# paste output here
```

</details>

### Relevant log output

```Shell

```

### PR DESCRIPTION
**Pull request by ali-a-a**
This PR tries to fix #19752. It adds support to authorization headers containing the keyword `Bearer`. It would let proxies expecting bearer headers work with etcd APIs.

### CODE REVIEW

File: server/auth/store_test.go [hunk 1]
```diff
@@ -836,12 +836,30 @@ func TestAuthInfoFromCtx(t *testing.T) {
...
 		t.Errorf("expected %v, got %v", ErrInvalidAuthToken, err)
 	}
 
+	ctx = metadata.NewIncomingContext(t.Context(), metadata.New(map[string]string{rpctypes.TokenFieldNameGRPC: bearerPrefix + "Invalid.Token"}))
+	_, err = as.AuthInfoFromCtx(ctx)
+	if !errors.Is(err, ErrInvalidAuthToken) {
+		t.Errorf("expected %v, got %v", ErrInvalidAuthToken, err)
+	}
+
+	ctx = metadata.NewIncomingContext(t.Context(), metadata.New(map[string]string{rpctypes.TokenFieldNameGRPC: bearerPrefix}))
```
Comment (zaitcev) [inline server/auth/store_test.go:857]: Is it really necessary to add both a test with bearerPrefix (with trailing whitespace) and a hardcoded "Bearer" (with no trailing whitespace) at line +839? We already have tickets for looking into tests running too slowly.

File: server/auth/store.go [hunk 2]
```diff
@@ -1072,6 +1074,12 @@ func (as *authStore) AuthInfoFromCtx(ctx context.Context) (*AuthInfo, error) {
 	}
 
 	token := ts[0]
+
+	// support authorization headers with bearer prefix.
+	if strings.HasPrefix(token, bearerPrefix) {
+		token = strings.Split(token, bearerPrefix)[1]
```
Comment (zaitcev) [inline server/auth/store.go]: Ideomatically I prefer
		token = token[len(bearerPrefix):]

Imagine a situation in the very far future when the token happens to include a string "Bearer " verbatim. It is impossible today because 0x20 space is not in base64 alphabet, but why deviate from a safe practice?

**Review by zaitcev — APPROVED** [review]
